src/ProductConstructionService/ProductConstructionService.DependencyFlow/WorkItemProcessors/SubscriptionUpdateProcessor.cs

```diff
@@ -15,7 +15,7 @@ public override async Task<bool> ProcessWorkItemAsync(
         CancellationToken cancellationToken)
     {
         var updater = _updaterFactory.CreatePullRequestUpdater(PullRequestUpdaterId.Parse(workItem.UpdaterId));
-        await updater.ProcessPendingUpdatesAsync(workItem, forceApply: false);
+        await updater.ProcessPendingUpdatesAsync(workItem, forceApply: false, build: null);
```

Comment:
This is the only place where we pass `null` as `build` if I am correct.
Why not just make the parameter required and query it here based on `workItem.BuildId`?
I think it will make the API simpler.